File: src/main/java/com/tom/guess/Game.java
package com.tom.guess;

import java.util.Random;
import java.util.Scanner;

public class Game {
    public static void main(String[] args){

        int secret = new Random().nextInt(100)+1;
        Scanner scanner = new Scanner(System.in);
        byte number = 0;
        while(number != secret){
            System.out.println("Please enter number: ");
            number = scanner.nextByte();
            if (number > secret){
                System.out.println("lower");
            }else if (number < secret){
                System.out.println("higher");
            }else {
                System.out.println("Great, a number is " + secret);
            }
        }
    }
}

File: src/main/kotlin/com/tom/guess/Game.kt
package com.tom.guess

import java.util.*

fun main(agrs: Array<String>){
    val secret = Random().nextInt(100).toByte()
    val scanner = Scanner(System.`in`)
    var number : Byte = -1
    while (number != secret){
        print("Please enter a number: ")
        number = scanner.nextByte()
        when{
            number > secret -> println("lower")
            number < secret -> println("higher")
            else -> println("Great, number is $secret")
        }
    }

}